src/main/java/com/example/springresttask/service/EmailService.java

```diff
@@ -23,16 +27,17 @@ public Email createEmailDelivery(Email email) {
     }
 
     @Transactional
-    public Email updateDeliveryDate(Email email) {
-        if (!emailRepository.existsById(email.getId())) {
-            throw new RuntimeException("id not found");
-        }
+    public EmailDto updateDeliveryDate(Long id, EmailDto emailDto) {
+        Email email1 = emailRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Email not found!"));
 
-        if (email.getIsSent()) {
+        if (emailDto.getIsSent()) {
```

Comment:
Maybe you meant email? You would like to check against a persisted entity, not against what client sends you